### ISSUE TITLE
Generated foreign key columns use table name instead of property name

### ISSUE THREAD

**Issue by jhaislip** [jdkandersson/OpenAlchemy#139]
According to the docs:

> When OpenAlchemy constructs the foreign key for you, the name of the property is made up of the name of the property referencing the object (division in the previous example) combined with the name of the property in the foreign key constraint (id in the previous example). These two values are combined as follows: `“<reference property name>_<foreign key column name>”` (division_id in the previous example).

However,  it appears to be using `<foreign key table name>_<foreign key column name>` which causes problems if a schema has two properties which reference the same object.   

Example:

```
Supervisor:
  type: object
  x-tablename: supervisor
  properties:
    id:
      type: integer
      description: Unique identifier for the supervisor.
      example: 0
      x-primary-key: true
      x-autoincrement: true
      readOnly: true
Employee:
  type: object
  x-tablename: employee
  properties:
    id:
      type: integer
      description: Unique identifier for the employee.
      example: 0
      x-primary-key: true
      x-autoincrement: true
      readOnly: true
    primary_supervisor:
      $ref: "#/components/schemas/Supervisor"
    secondary_supervisor:
      $ref: "#/components/schemas/Supervisor"
```

When I generate the Base using init_yaml and create the tables using Base.metadata.create_all(engine) I'm only seeing a single foreign key column in the employee table called `supervisor_id `instead of two separate columns called `primary_supervisor_id ` and `secondary_supervisor_id`. 

Is this the expected behavior?

**Comment by jdkandersson** [comment]
Thanks for reporting the problem. This will be resolved by:

- [ ] Adding an integration test based on your example
- [ ] Recording the logical name of a property (its name in the schema) in an object in the gathered artefacts
- [ ] Changing the name of the foreign key property to be based on the logical name

**Comment by jdkandersson** [comment]
There is a pull request that fixes this bug. However, there are some other things that need to be done before the above use case will work. I will add this as a feature request. The problem is that, currently, the `SQLAlchemy` `relationship` function is responsible for determining which foreign key to use for a particular relationship. This only works where there is a single relationship between any two tables, which is not the case above. There is a work around that will allow this use case to work. The example `OpenAPI` spec would be:

```
Supervisor:
  type: object
  x-tablename: supervisor
  properties:
    id:
      type: integer
      description: Unique identifier for the supervisor.
      example: 0
      x-primary-key: true
      x-autoincrement: true
      readOnly: true
Employee:
  type: object
  x-tablename: employee
  properties:
    id:
      type: integer
      description: Unique identifier for the employee.
      example: 0
      x-primary-key: true
      x-autoincrement: true
      readOnly: true
    primary_supervisor:
      allOf:
        - $ref: "#/components/schemas/Supervisor"
        - x-kwargs:
            foreign_keys: Employee.primary_supervisor_id
    secondary_supervisor:
      allOf:
        - $ref: "#/components/schemas/Supervisor"
        - x-kwargs:
            foreign_keys: Employee.secondary_supervisor_id
```

I haven't specifically tested the above `YAML` so there might be some problems in it. However, there is a test case that illustrates this which you can see here: [https://github.com/jdkandersson/OpenAlchemy/blob/a7ee03c796b13ef9508e11a02402ac38fa61ea97/tests/open_alchemy/integration/test_database/test_relationship.py#L541](https://github.com/jdkandersson/OpenAlchemy/blob/a7ee03c796b13ef9508e11a02402ac38fa61ea97/tests/open_alchemy/integration/test_database/test_relationship.py#L541)

**Comment by jdkandersson** [comment]
This has now been released in version 1.3.0.
